merkletree/proof_test.go

```diff
@@ -7,6 +7,16 @@ import (
 	"github.com/coniks-sys/coniks-go/utils"
 )
 
+func verifyAuthPath(ap *AuthenticationPath, key, value []byte, treehash []byte) bool {
```

Comment:
I'm wondering if, rather than inlining the `ap.Verify` and `tb.Verify` and ending up w/ a duplicated `verifyAuthPath`, it would be better to change the return type of those methods to `error`?